### PR TITLE
security fix: bump pg to 6.1.6

### PR DESCRIPTION
**Pull request by shackbarth**
See
https://snyk.io/vuln/npm:pg:20170813
https://node-postgres.com/announcements#2017-08-12-code-execution-vulnerability

Seems like a pretty safe bump. I tried this new version out on my own sails app and all the tests still pass.

### CODE REVIEW

**Comment by danil-z** [comment]
please merge it

**Comment by fenichelar** [comment]
@mikermcneil Thoughts? This is causing a lot of security alerts.

**Comment by mikermcneil** [comment]
@shackbarth thank you for testing this against your app, that helps a ton!  As soon as I get a moment, I'll merge, do another test w/ sails-postgresql, and then hopefully be able to publish this package.

@fenichelar @danil-z Hey y'all, as far as I know, this is pretty isolated.  My understanding is that, in order for this to be an issue for your app, you'd have to either (A) be your own attacker and set a deliberately dangerous `columnName` in your model definition or (B) use a native query where you're using untrusted user data as the column name.  Here's what I'm basing that off of:

image

The reason I point this out is that, in the current ecosystem, developers are in an endless battle against vulnerability-sniffing CI tools that test for vulnerabilities using blind package.json tracing without proper context for whether a particular vulnerability in a dependency might or might not be relevant for a userland package or application, or _whether it is even in a code path that the userland package or dependency uses_.

`<rant>`
Every time this happens, it encourages package maintainers and app developers to drop dependencies, which just raises the chances for security vulnerabilities as we all re-implement things in a bazillion different ways just to avoid the long-term maintenance burden of dealing with automatically-generated vulnerability notices.

tldr; with this kind of stuff, always please let me know if I'm missing something and this is actually critical.  But please also come into it with a healthy dose of skepticism, because so many of these are completely irrelevant.
`</rant>`

Thanks so much for your help, guys.  And sorry for the long-winded follow up.

**Comment by fenichelar** [comment]
@mikermcneil Thanks!

> The reason I point this out is that, in the current ecosystem, developers are in an endless battle against vulnerability-sniffing CI tools that test for vulnerabilities using blind package.json tracing without proper context for whether a particular vulnerability in a dependency might or might not be relevant for a userland package or application, or whether it is even in a code path that the userland package or dependency uses.
> Every time this happens, it encourages package maintainers and app developers to drop dependencies, which just raises the chances for security vulnerabilities as we all re-implement things in a bazillion different ways just to avoid the long-term maintenance burden of dealing with automatically-generated vulnerability notices.

This is a very good point! One thing I will mention, I think the pinning of specific package versions makes Sails and the related Sails packages especially hard to maintain from this perspective. I completely understand the reason pinning was chosen. However, it seems to me that more and more packages are focusing on making sure to maintain backwards compatibility with upgrades. I'm having less and less regression issues every year. It might be time to reconsider package pinning? Just a thought :)

As always, Sails is awesome and your work is greatly appreciated!!

**Comment by mikermcneil** [comment]
@fenichelar you are totally right.  I keep thinking the same thing, and then years of scar tissue keep preventing me from doing it.  Keep working on me, I can be turned eventually

**EDIT:**  at least for certain packages

**Comment by fenichelar** [comment]
@mikermcneil It may make sense to start slow. For the large, stable packages upin instead of update the pin going forward. We could even start here... `pg` is a large, stable package... ;)

**Comment by shackbarth** [comment]
Don't give in to temptation! I've got all the same scar tissue and the world is no safer in 2018. I really value the assurance that my same app that built yesterday will still build today. I realize it adds a maintenance burden to all of those package owners that don't cheat by using the `~`, but really, that cheat is JS cowboy insanity. Greenkeeper and their ilk work pretty well to mitigate the pain, from what I've heard.

I also totally understand about snyk fatigue. I'm riding a dozen snyk alerts that have no possible bearing on my app. But this alert actually does, so here's my report from userland:

We let some users execute a bespoke SQL codeblock of their own authorship against the database. SQL Injection As A Service, I know sounds nuts, but these are users that would otherwise have a direct login to the database. So we guard that privilege with the same zealousness that we would guard giving out a well-granted database account (we keep the list in an OS env var). But we do *not* guard that privilege with the same zealousness that we would guard giving out an OS login. It's not a huge deal because these are users that we trust quite a lot, but it's still a form of privilege escalation.

**Comment by mikermcneil** [comment]
@shackbarth @fenichelar @danil-z Thanks all!
